### PR TITLE
Switch to regatta-go client

### PR DESCRIPTION
**Pull request by coufalja**
Draft of switch to official Go client
* Less dependencies
* Automatic TLS, Connection and Compression management

### CODE REVIEW

File: cmd/delete.go [hunk 1]
```diff
@@ -20,50 +20,39 @@ var Delete = cobra.Command{
...
 		key = strings.TrimSuffix(key, "*")
 		if len(key) == 0 {
 			// delete all
-			return &regattapb.DeleteRangeRequest{
-				Table:    []byte(table),
-				Key:      []byte{0},
-				RangeEnd: []byte{0},
-				PrevKv:   true,
-			}
+			return string(zero), []client.OpOption{client.WithRange(string(zero)), client.WithPrevKV()}
```
Comment (Tantalor93) [inline cmd/delete.go:51]: now whenever constant `zero` is used it is always used as string, could you change constant `zero` to string?

```
	zero = string([]byte{0})
```
and simplify usage?

File: cmd/delete.go [hunk 2]
```diff
@@ -20,50 +20,39 @@ var Delete = cobra.Command{
...
+	Run: func(cmd *cobra.Command, args []string) {
 		timeoutCtx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
 		defer cancel()
-		req := createDeleteRangeRequest(args)
-
-		_, err = client.DeleteRange(timeoutCtx, req)
+		key, opts := keyAndOptsForDelete(args)
+		_, err := regatta.Table(args[0]).Delete(timeoutCtx, key, opts...)
 		if err != nil {
 			handleRegattaError(cmd, err)
```
Comment (Tantalor93) [inline cmd/delete.go:40]: just a question: does regatta-go client return GRPC errors still? That is little unexpected, would expect it to be abstracted away

File: cmd/regatta.go [hunk 3]
```diff
@@ -1,42 +1,32 @@
...
+type logger struct{}
 
-	connOpts := []grpc.DialOption{
-		// nolint:gosec
-		grpc.WithTransportCredentials(credentials.NewTLS(&tls.Config{RootCAs: pool, InsecureSkipVerify: insecureOption})),
-	}
+func (p logger) Infof(s string, a ...any)  {}
+func (p logger) Debugf(s string, a ...any) {}
+func (p logger) Warnf(s string, a ...any)  {}
+func (p logger) Errorf(s string, a ...any) {}
```
Comment (Tantalor93) [inline cmd/regatta.go:15]:
```suggestion
func (p logger) Infof(_ string, _ ...any)  {}
func (p logger) Debugf(_ string, _ ...any) {}
func (p logger) Warnf(_ string, _ ...any)  {}
func (p logger) Errorf(_ string, _ ...any) {}
```

File: cmd/put.go [hunk 4]
```diff
@@ -28,45 +23,41 @@ var Put = cobra.Command{
...
 	},
 }
 
-func createPutRequest(args []string) (*regattapb.PutRequest, error) {
-	table := []byte(args[0])
-	key := []byte(args[1])
-	var value []byte
+func keyAndValueForPut(args []string) (string, string, error) {
 	if putBinary {
 		var err error
```
Comment (Tantalor93) [inline cmd/put.go:55]: this err variable looks useless now, can we remove?

File: cmd/put_test.go [hunk 5]
```diff
@@ -2,37 +2,27 @@ package cmd
...
-	go s.Serve(lis)
-	defer s.Stop()
+	mclient := &mockClient{}
+	regatta = mclient
+	mclient.On("Table", "table").Return(mtbl)
 
 	buf := new(bytes.Buffer)
 	RootCmd.SetOut(buf)
-	RootCmd.SetArgs([]string{"--endpoint", lis.Addr().String(), "--cert", "test.crt", "put", "table", "key", "data"})
+	RootCmd.SetArgs([]string{"--endpoint", "", "--cert", "test.crt", "put", "table", "key", "data"})
```
Comment (Tantalor93) [inline cmd/put_test.go:22]: it seems I am missing some validation of argument, as passing empty `endpoint` should not be possible 🤔

File: cmd/utils_test.go [hunk 6]
```diff
@@ -2,58 +2,41 @@ package cmd
...
-	return called.Get(0).(*regattapb.PutResponse), called.Error(1)
+func (m *mockTable) Delete(ctx context.Context, key string, opts ...client.OpOption) (*client.DeleteResponse, error) {
+	args := m.Called(ctx, key, opts)
+	return args.Get(0).(*client.DeleteResponse), args.Error(1)
 }
 
-func (m *mockKVService) Delete(ctx context.Context, req *regattapb.DeleteRangeRequest) (*regattapb.DeleteRangeResponse, error) {
-	called := m.Called(ctx, req)
-	return called.Get(0).(*regattapb.DeleteRangeResponse), called.Error(1)
+func (m *mockTable) Txn(ctx context.Context) client.Txn {
```
Comment (Tantalor93) [inline cmd/utils_test.go:35]:
```suggestion
func (m *mockTable) Txn(_ context.Context) client.Txn {
```

File: cmd/delete.go [hunk 7]
```diff
@@ -20,50 +20,39 @@ var Delete = cobra.Command{
...
+	PreRun: func(cmd *cobra.Command, args []string) {
+		// Allow for mocking in tests.
+		if regatta == nil {
+			cl, err := createClient()
+			if err != nil {
+				cmd.PrintErrln("There was an error, while establishing connection to Regatta.", err)
+				return
+			}
+			regatta = cl
 		}
```
[the remark this answers is not in the document]
Comment (coufalja) [reply cmd/delete.go:32]: I thought about it but there is currently one and in the future maybe more commands that does not rely on connection being established (man command for now). I will check if there is a way to share that code some other way.

File: cmd/delete.go (hunk 2 above)
Comment (coufalja) [reply cmd/delete.go:26]: Currently yes, it abstacts context and timeout errors but other gRPC errors are still propagated, room for improvement there!

**Comment by Tantalor93** [comment]
LGTM, thx!

**Pull request merged**